pkg/cmd/adm/restart.go

```diff
@@ -3,155 +3,198 @@ package adm
 	command := &cobra.Command{
-		Use:   "restart -t <cluster-name> <deployment-name>",
-		Short: "Restarts a deployment",
-		Long: `Restarts the deployment with the given name in the operator namespace. 
-If no deployment name is provided, then it lists all existing deployments in the namespace.`,
+		Use:   "restart <cluster-name>",
+		Short: "Restarts an operator",
+		Long: `Restarts the whole operator in the given cluster name. 
+		It restarts the operator and checks the status of the deployment`,
 		Args: cobra.RangeArgs(0, 1),
 		RunE: func(cmd *cobra.Command, args []string) error {
 			term := ioutils.NewTerminal(cmd.InOrStdin, cmd.OutOrStdout)
 			ctx := clicontext.NewCommandContext(term, client.DefaultNewClient)
-			return restart(ctx, targetCluster, args...)
+			return restart(ctx, args...)
 		},
 	}
-	command.Flags().StringVarP(&targetCluster, "target-cluster", "t", "", "The target cluster")
-	flags.MustMarkRequired(command, "target-cluster")
 	return command
 }
 
-func restart(ctx *clicontext.CommandContext, clusterName string, deployments ...string) error {
+func restart(ctx *clicontext.CommandContext, clusterNames ...string) error {
+	if clusterNames == nil {
+		return fmt.Errorf("please provide a cluster name to restart the operator e.g `ksctl adm restart host`")
+	}
+	clusterName := clusterNames[0]
+	kubeConfigFlags := genericclioptions.NewConfigFlags(true).WithDeprecatedPasswordFlag()
+	factory := cmdutil.NewFactory(cmdutil.NewMatchVersionFlags(kubeConfigFlags))
+	ioStreams := genericclioptions.IOStreams{
+		In:     os.Stdin,
+		Out:    os.Stdout,
+		ErrOut: os.Stderr,
+	}
+	kubeConfigFlags.ClusterName = nil  // `cluster` flag is redefined for our own purpose
+	kubeConfigFlags.AuthInfoName = nil // unused here, so we can hide it
+	kubeConfigFlags.Context = nil      // unused here, so we can hide it
+
 	cfg, err := configuration.LoadClusterConfig(ctx, clusterName)
 	if err != nil {
 		return err
 	}
-	cl, err := ctx.NewClient(cfg.Token, cfg.ServerAPI)
+	kubeConfigFlags.Namespace = &cfg.OperatorNamespace
+	kubeConfigFlags.APIServer = &cfg.ServerAPI
+	kubeConfigFlags.BearerToken = &cfg.Token
+	kubeconfig, err := client.EnsureKsctlConfigFile()
 	if err != nil {
 		return err
 	}
+	kubeConfigFlags.KubeConfig = &kubeconfig
 
-	if len(deployments) == 0 {
-		err := printExistingDeployments(ctx.Terminal, cl, cfg.OperatorNamespace)
-		if err != nil {
-			ctx.Terminal.Printlnf("\nERROR: Failed to list existing deployments\n :%s", err.Error())
-		}
-		return fmt.Errorf("at least one deployment name is required, include one or more of the above deployments to restart")
+	cl, err := ctx.NewClient(cfg.Token, cfg.ServerAPI)
+
+	if err != nil {
+		return err
 	}
-	deploymentName := deployments[0]
 
 	if !ctx.AskForConfirmation(
-		ioutils.WithMessagef("restart the deployment '%s' in namespace '%s'", deploymentName, cfg.OperatorNamespace)) {
+		ioutils.WithMessagef("restart the '%s' operator in namespace '%s'", clusterName, cfg.OperatorNamespace)) {
```

Comment:
should this be something like:
```suggestion
		ioutils.WithMessagef("restart all the deployments in the cluster '%s' and namespace '%s'", clusterName, cfg.OperatorNamespace)) {
```